exo-run/src/app-runner.ls

```diff
@@ -56,17 +56,32 @@ class AppRunner extends EventEmitter
     DockerCompose.kill-all-containers {cwd: @docker-config-location, @write}, -> process.exit exit-code
 
 
-  _compile-online-text: ->
-    online-texts = {}
+  _compile-online-text: (done) ~>
+    @online-texts = {}
     for app-dependency in @app-config.dependencies
       dependency = ApplicationDependency.build app-dependency
-      online-texts[app-dependency.type] = dependency.get-online-text!
+      @online-texts[app-dependency.type] = dependency.get-online-text!
+    services = []
     for protection-level of @app-config.services
       for role, service-data of @app-config.services[protection-level]
-        if service-data.location #TODO: compile online text for external services
-          service-config = yaml.safe-load fs.read-file-sync(path.join(process.cwd!, service-data.location, 'service.yml'))
-          online-texts[role] = service-config.startup['online-text']
-    online-texts
+        services.push {role: role, service-data: service-data}
+    async.map-series services, @_get-online-text, (err) ~>
+      | err => done err
+      done!
+
+
+  _get-online-text: ({role, service-data}, done) ~>
+    | service-data.location
```

Comment:
This is missing a `=>`, though apparently that's not required... Let's add it just to stay consistent